packages/typegpu/src/core/texture/externalTexture.ts

```diff
@@ -25,28 +32,59 @@ export function isExternalTexture<T extends TgpuExternalTexture>(
 
 export class TgpuExternalTextureImpl
   implements TgpuExternalTexture, SelfResolvable {
-  public readonly resourceType = 'external-texture';
-  public readonly [$wgslDataType]: AnyData;
+  readonly [$runtimeResource] = true;
+  readonly [$wgslDataType] = {} as WgslExternalTexture;
```

Comment:
Since these are not being used as runtime values anymore, these should be irrelevant
```suggestion
```